src/common/expression/LogicalExpression.h

```diff
@@ -79,6 +79,15 @@ class LogicalExpression final : public Expression {
 
   bool isLogicalExpr() const override { return true; }
 
+  void setLogicalKind(Kind kind) {
+    if ((kind == Expression::Kind::kLogicalOr || kind == Expression::Kind::kLogicalAnd) &&
+        (kind_ == Expression::Kind::kLogicalOr || kind_ == Expression::Kind::kLogicalAnd)) {
```

Comment:
following check maybe better to understand:
```c++
(kind == Expression::Kind::kLogicalOr && kind_ == Expression::Kind::kLogicalAnd) ||
        (kind == Expression::Kind::kLogicalAnd && kind_ == Expression::Kind::kLogicalOr)
```

Reply:
and this function could be renamed: `reverseLogicalKind()`

Reply:
```c++
void reverseLogicalKind() {
  if (kind_ == kLogicalAnd) {
    kind_ = kLogicalOr;
  } else if (kind_ == kLogicalOr) {
    kind_ = kLogicalAnd;
  } else {
    LOG(FATAL) << "Should not reverser logical expression except and/or kind.";
  }
}
```

Reply:
That's OK